lib/spack/docs/build_systems/luapackage.rst

```diff
@@ -0,0 +1,93 @@
+.. Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _luapackage:
+
+------------
+LuaPackage
+------------
+
+LuaPackage is a helper for the common case of Lua packages that provide
+a rockspec file.  This is not meant to take a rock archive, but to build
+a source archive or repository that provides a rockspec, which should cover
+most lua packages. In the case a Lua package builds by Make rather than
+luarocks, prefer MakefilePackage.
+
+^^^^^^
+Phases
+^^^^^^
+
+The ``LuaPackage`` base class comes with the following phases:
+
+#. ``preprocess`` - generate ninja files
```

Comment:
Are there any examples of packages that use this (or will need this) phase? Or is this just hypothetical?

Reply:
I had one that needed it, but reworked it using a newer luarocks dependency.  There are a *lot* of old largely unmaintained but critical lua packages around, I expect it to come up every so often.

Reply:
This is actually back in active use, the lua-bitlib package needs it to fix the `luaL_reg` calls to be capitalized correctly.